indexer/layerscanner/layerscanner.go

```diff
@@ -235,9 +237,16 @@ func (r *result) Do(ctx context.Context, s indexer.VersionedScanner, l *claircor
 	default:
 		panic(fmt.Sprintf("programmer error: unknown type %T used as scanner", s))
 	}
-	if err != nil {
+	addrErr := &net.AddrError{}
+	switch {
+	case errors.Is(err, nil):
+	case errors.As(err, &addrErr):
```

Comment:
Could get more specific here if needed

Reply:
I don't think that's needed immediately.